### DISCUSSION TITLE
Task 3

### DISCUSSION THREAD

**Discussion by novaz03** [Part-II-Computational-Physics/.github#21] [General]
Hi guys,

It appears I cannot get animation output for this Spectrum-related code.
I wonder if it is related with the explicit euler approach for ode (but it should not affect output of any sort?)

```python
k = np.fft.fftfreq(n_points_space,int(x_max-x_min))


def Spe(v, n_points_space, dx, dt):
    f_k = f.copy()
    #initialising the FT of f(x) to k space
    def step(f):
        f_k = np.fft.fft(f)
        #FFT of f into k space4
        for i in range(n_points_space): 
            f_k_init = 0
            f_k_fin = 0
            #dummy variables for holding the initial value of f_k_n and its final value
            #for each f_k_i:
            step_size = dt/10
            f_k_init = f_k[i]
            f_k_fin = f_k_init
            t = np.arange(0,dt+step_size,step_size)
            for j in range(len(t)-1):
                f_k_fin += h*(delta_f_k(f_k_fin,h*j))

            f_k[i] = f_k_fin
            f[i] = np.fft.ifft(f_k[i])
            print('1')
    return step
```

**Comment by NathanMagnan** [comment]
Hey,

Does the error measurement code work for this spectral code?
I'm asking because I want to know if your spectral scheme compiles. It's quite complex compared to the 3-line solution. 
I'm especially unsure about the `f[i] = np.fft.ifft(f_k[i])` line. It looks like you are applying the inverse fourier transform on a scalar instead of a vector. I also don't know what role the loop on `j` plays.
